diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/classes/DataClassesRule.kt

```diff
@@ -63,9 +64,14 @@ class DataClassesRule(private val configRule: List<RulesConfig>) : Rule("data-cl
 
     @Suppress("UnsafeCallOnNullableType", "FUNCTION_BOOLEAN_PREFIX")
     private fun ASTNode.canBeDataClass(): Boolean {
-        if (findChildByType(PRIMARY_CONSTRUCTOR)?.let { constructor -> (constructor.psi as KtPrimaryConstructor).valueParameters.none { it.hasValOrVar() } } == true) {
+        val isNotPropertyInClassBody = findChildByType(CLASS_BODY)?.let { (it.psi as KtClassBody).properties.isEmpty() } ?: true
+        val isNotPropertyInConstructor = findChildByType(PRIMARY_CONSTRUCTOR)
```

Comment:
```suggestion
        val hasPropertyInConstructor = findChildByType(PRIMARY_CONSTRUCTOR)
```
Very confusing name, this variable is true if there are only properties in constructor, right?

Reply:
Yes